### DISCUSSION TITLE
Welcome! The FAQ is here!

### DISCUSSION THREAD

**Discussion by berrystephenw** [berrystephenw/OctoText#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

The posts in this discussion have solutions to the most common problems found when setting up OctoText.

**Comment by berrystephenw** [comment]
## Quota's and Outlook app password
One issue that will likely come up with anyone that uses this plugin is quota's. Microsoft has a "flexible" quota system where they might decide to shut you off if you are new and their algorithms decide you might be spamming people.
The way around this restriction is to add security to your account. Add two factor authorization will get your account running again, but it means getting an API key for OctoText. It's easy to do, first go to the security tab in your account, then select advanced options:
![image](https://user-images.githubusercontent.com/44931130/111923022-87d9c480-8a73-11eb-975b-7248f2936688.png)
After you turn on two step, you will need to generate an app password:
![image](https://user-images.githubusercontent.com/44931130/111923082-bbb4ea00-8a73-11eb-946d-21563fe9b785.png)
An app password is nothing more than a password Microsoft generates for you. Use that password in the Octotext settings page:
![image](https://user-images.githubusercontent.com/44931130/111923129-f880e100-8a73-11eb-91ce-e26c0b9e2280.png)

**Comment by berrystephenw** [comment]
## Gmail issues
A few people are trying to get Gmail working with OctoText and having a hard time. My research (yep, I googled it) says that gmail is requiring app passwords for access to email for "less secure" devices. To set this up you should follow these directions:

https://support.google.com/accounts/answer/185833?hl=en

**Reply by subatomicglue** [reply]
This is not working for me.  Anyone have luck? 
I generated a new app password.   I also tried using an existing known working app password that I use for my HP printer.     

**host**: `smtp.gmail.com`
**port**: `465`
**Checking this box will force SSL encryption**:  `[checked]`
**Username and Host address**: `<username>@gmail.com`    (`<username>` in 1st field;  `gmail.com` in 2nd field)
**Email password**:   `<gmail generated app password>`
**Phone number OR user name @ Gateway OR Email address**:  `<username>@gmail.com`   (`<username>` in 1st field;  `gmail.com` in 2nd field)
**CC**:


Some working example here would be really helpful.

**Reply by berrystephenw** [reply]
Please see this discussion: https://github.com/berrystephenw/OctoText/discussions/127#discussion-4134709

The short answer is that gmail security has changed and requires a change in Octotext (which I don't have time for this summer).

**Reply by subatomicglue** [reply]
<img width="1462" alt="image" src="https://user-images.githubusercontent.com/7660853/184179781-43b04b26-4b4a-4eca-8464-873fd3507370.png">
uh, I think it's working.   after printing just now, I got these emails.   It's just that the test button failed with red text popup right after I set it up...      

- now, the "mail" button in the top results in a green text popup (success)
- now, the test button in the setup window also results in a green text popup (success)

What changed?

- Maybe this: I **did** reply to an email that gmail.com sent me asking if it was me who created the app password, I clicked the "Yes, this was me" button they presented...     Weirdly, I got the fail on the (long time used) HP Printer app password, but maybe gmail filters the sending device's ID somehow (?) in their security checks... ?  (i did just verify the HP printer is still sending emails, so that app pass is still good)

**Reply by berrystephenw** [reply]
I'm glad you were able to get it working - thanks for sharing your solution!

**Reply by MatzeMuc86** [reply]
Sorry but still not working with gmail following this information and using app specific password. Any news?

**Reply by callport** [reply]
I was experiencing some of this difficulty today, as well.  However, in my email profile (setting up a GMail Relay), there is a setting "LEss secure app access" -- as soon as I enabled this, everything worked.

![image](https://github.com/berrystephenw/OctoText/assets/959016/fda6ae4f-612e-4737-9600-1e760774569b)

**Reply by wjcarpenter** [reply]
FWIW, I tried about a million different things today, including all the stuff mentioned in this thread. No joy. To take OctoText out of the loop, I switched to experimenting via Thunderbird. Same failed results. I haven't used SMTP with Gmail for years, so maybe it just doesn't work if you're not using their enterprise paid version or something.

**Reply by justgosh** [reply]
Don't put in the spaces and it works.

**Comment by berrystephenw** [comment]
## Installation is Incompatible 
If you are trying to install OctoText and you see a message that looks like this:

![image](https://user-images.githubusercontent.com/44931130/113294585-2b717300-92c5-11eb-930d-12723f46b0c5.png)

then you are running an older version of Python. OctoText relies on Python 3 for some of the built in libraries that it uses and the installation will fail if it doesn't meet this requirement.

The fix for this, if your other plugins will allow (please check to make sure they are Python 3 compatible), is to upgrade your Octoprint distribution. For the instructions and an explanation of why this is necessary please see this excellent blog post:
https://community.octoprint.org/t/upgrade-your-octoprint-install-to-python-3/23973

**Reply by berrystephenw** [reply]
## A list of international SMS gateways

https://sms-gateway-service.com/sms-gateway-by-country/
